unity-client/Assets/ABConverter/VisualTests.cs

```diff
@@ -0,0 +1,220 @@
+                    Debug.Log("Visual Test Detection: FAILED converting asset -> " + go.name);
+                }
+
+                go.SetActive(false);
+            }
+
+            VisualTestHelpers.baselineImagesPath = baselinePath;
+            VisualTestHelpers.testImagesPath = testImagesPath;
+
+            OnFinish?.Invoke ((core != null) ? core.state.lastErrorCode : Core.ErrorCodes.UNDEFINED);
+            yield break;
+        }
+
+        public static GameObject[] LoadAndInstantiateAllGltfAssets()
+        {
+            var assets = AssetDatabase.FindAssets($"t:GameObject", new[] {"Assets/_Downloaded"});
+
+            List<GameObject> importedGLTFs = new List<GameObject>();
+
+            foreach (var guid in assets)
+            {
+                GameObject gltf = AssetDatabase.LoadAssetAtPath<GameObject>(AssetDatabase.GUIDToAssetPath(guid));
+                var importedGLTF = Object.Instantiate(gltf);
+                importedGLTF.name = importedGLTF.name.Replace("(Clone)", "");
+                importedGLTFs.Add(importedGLTF);
+            }
+
+            return importedGLTFs.ToArray();
+        }
+
+        public static GameObject[] LoadAndInstantiateAllAssetBundles()
+        {
+            Caching.ClearCache();
+
+            string workingFolderName = "_Downloaded";
+
+            var pathList = Directory.GetDirectories(Application.dataPath + "/" + workingFolderName);
+
+            List<string> dependencyAbs = new List<string>();
+            List<string> mainAbs = new List<string>();
+
+            foreach (var paths in pathList)
+            {
+                var hash = new DirectoryInfo(paths).Name;
+                var path = "Assets/" + workingFolderName + "/" + hash;
+                // var guids = AssetDatabase.FindAssets("t:GameObject", new[] {path});
+                var guids = AssetDatabase.FindAssets("t:GameObject", new[] {path});
+
+                // NOTE(Brian): If no gameObjects are found, we assume they are dependency assets (textures, etc).
+                if (guids.Length == 0)
+                {
+                    dependencyAbs.Add(hash);
+                }
+                else
+                {
+                    // Otherwise we assume they are gltfs.
+                    mainAbs.Add(hash);
+                }
+            }
+
+            // NOTE(Brian): We need to store the asset bundles so they can be unloaded later.
+            List<AssetBundle> loadedAbs = new List<AssetBundle>();
+
+            foreach (var hash in dependencyAbs)
+            {
+                string path = abPath + hash;
+                var req = UnityWebRequestAssetBundle.GetAssetBundle(path);
+
+                if(SystemInfo.operatingSystemFamily == OperatingSystemFamily.MacOSX)
+                    req.url = req.url.Replace("http://localhost", "file:///");
+
+                req.SendWebRequest();
+
+                while (!req.isDone)
+                {
+                }
```

Comment:
Can this be move to a Courutine?

Reply:
The problem with that is that the `LoadAndInstantiateAllAssetBundles()` needs to return the collection of instantiated gameobjects (`GameObject[]`), to make the change you suggest we should change that method to be a coroutine, thus returning an `IEnumerator`, and have the array pass as a reference parameter or something similar.
Since all the Asset Bundle Conversor code runs in editor more (in a service we have running and converting assets) and not in production, I wouldn't make this change you suggest, otherwise i think you are right and we should turn these `while(true)` waits into coroutines to avoid stopping the main thread.